src/lib/types.hpp

```diff
@@ -29,17 +29,13 @@
  * like this
  *   ChunkID x = 3;
  * but need to use
- *   ChunkID x{3}; or
  *   auto x = ChunkID{3};
+ * In some cases (e.g., when narrowing data types), casting to the base_type first might be necessary, e.g.:
+ *   ChunkID{static_cast<ChunkID::base_type>(size_t{17})}
  *
- * WorkerID, TaskID, CommitID, and TransactionID are used in std::atomics and
- * therefore need to be trivially copyable. That's currently not possible with
- * the strong typedef (as far as I know).
- *
- * TODO(anyone): Also, strongly typing ChunkOffset causes a lot of errors in
- * the group key and adaptive radix tree implementations. Unfortunately, I
- * wasn’t able to properly resolve these issues because I am not familiar with
- * the code there
+ * We prefer strong typedefs, whenever they are applicable and make sense. However, there are cases where we cannot
+ * directly use them. For example in std::atomics as they need to be trivially copyable. Therefore, we use the
```

Comment:
I have to think about that again. I am not sure the copyable part is what bothers us here, but that we want the specialization for integral types (e.g. `++atomic_var`).